test/scenarios/default.py

```diff
@@ -1,100 +1,41 @@
-### Tests
+def electrs():
+    machine.wait_for_open_port(4224)  # prometeus metrics provider
 
-assert_running("setup-secrets")
-# Unused secrets should be inaccessible
-succeed('[[ $(stat -c "%U:%G %a" /secrets/dummy) = "root:root 440" ]]')
 
-assert_running("bitcoind")
-machine.wait_until_succeeds("bitcoin-cli getnetworkinfo")
-assert_matches("su operator -c 'bitcoin-cli getnetworkinfo' | jq", '"version"')
-# Test RPC Whitelist
-machine.wait_until_succeeds("su operator -c 'bitcoin-cli help'")
-# Restating rpcuser & rpcpassword overrides privileged credentials
-machine.fail(
-    "bitcoin-cli -rpcuser=publicrpc -rpcpassword=$(cat /secrets/bitcoin-rpcpassword-public) help"
-)
-machine.wait_until_succeeds(
-    log_has_string("bitcoind", "RPC User publicrpc not allowed to call method help")
-)
+def spark_wallet():
+    machine.wait_for_open_port(9737)
+    spark_auth = re.search("login=(.*)", succeed("cat /secrets/spark-wallet-login"))[1]
```

Comment:
Is there a reason why you changed the order of `machine.wait_for_open_port` & `spark_auth`?

Reply:
yes, you usually define variables when you use them and not before

Reply:
ACK